centreon/www/class/centreonGraph.class.php

```diff
@@ -456,20 +455,24 @@ public function initCurveList()
 
         /* Manage reals metrics */
         if (isset($l_rselector)) {
-            $DBRESULT = $this->DBC->query(
+            $statement = $this->DBC->prepare(
                 "SELECT host_id, service_id, metric_id, metric_name, unit_name, replace(format(warn,9),',','') warn,
-                    replace(format(crit,9),',','') crit
-                    FROM metrics AS m, index_data AS i
-                    WHERE index_id = id
-                    AND " . $l_rselector . "
-                    AND m.hidden = '0'
-                    ORDER BY m.metric_name"
+                replace(format(crit,9),',','') crit
+                FROM metrics AS m, index_data AS i
+                WHERE index_id = id
+                AND :l_rselector
```

Comment:
Doesn't work because the 'bindValue' function will enclose the value in single quotes.

Reply:
`SELECT host_id, service_id, metric_id, metric_name, unit_name, replace(format(warn,9),',','') warn,
                replace(format(crit,9),',','') crit
FROM metrics AS m, index_data AS i
WHERE index_id = id
AND 'metric_id IN (\'1\')'
AND m.hidden = '0'
ORDER BY m.metric_name`